src/qt/window.cpp

```diff
@@ -1387,12 +1388,19 @@ bool wxWindowQt::QtHandleMouseEvent ( QWidget *handler, QMouseEvent *event )
             wxFAIL_MSG( "Unknown mouse event type" );
     }
 
-    // Fill the event
-    QPoint mousePos = event->pos();
+    // Fill the event - convert screen position to the window's coordinates
+    wxPoint mousePos = ScreenToClient(wxQtConvertPoint(event->globalPos()));
+
+    // Correct the mouse position if we have a caption
+    if (HasFlag(wxCAPTION))
```

Comment:
Hmm, will this still be needed after #1402 will have been applied?

Reply:
I assume so as most of our changes have been merged into our [gmd_consolidation_branch](https://github.com/GeoTeric/wxWidgets/commits/gmd_consolidation_branch), which didn't cause a conflict and must have been an additional requirement.

Reply:
Can you please check? It does sound like it should not be needed once the client area is correctly calculated (or maybe `ScreenToClient()` still has a bug in it?)

Reply:
I've tried removing this code in our consolidation branch and found that it is still necessary to be able to un-dock aui panes, like the Tree Pane in the 'auidemo' sample. I believe this also caused an issue with being able to click on the close button of aui panes if they were initialised to an un-docked state.

Reply:
I don't really understand the reference to AUI here (was that comment intended for this PR?), but to rephrase the question: are the change in #1402 part of your _consolidation branch_?

Reply:
Yes, apologies if it was unclear but both this change and #1402 are part of our consolidation branch.

The AUI comment was what happens if I try removing the `if (HasFlag(wxCAPTION))` section combined with #1402.

Reply:
Yes, apologies if it was unclear but both this change and #1402 are part of our consolidation branch.

The AUI comment was what happens if I try removing the 'if (HasFlag(wxCAPTION))' section combined with #1402.

Reply:
But then, if there is a problem with AUI, shouldn't it be dealt with in AUI-specific code?
My concern here is about native TLWs, i.e. `wxFrame` and `wxDialog` - what happens for them? Is that "correction" required or not for them?

Reply:
I think this change was added here as the AUI code is largely generic and not split for different versions. I personally don't know of any other usages of `wxCAPTION` where this adjustment may or may not be necessary but it's certainly not present for all TLWs. Up to you whether you think it should come in I guess.

Reply:
> I personally don't know of any other usages of `wxCAPTION` where this adjustment may or may not be necessary but it's certainly not present for all TLWs. Up to you whether you think it should come in I guess.

Ok, there are 2 things that do not add up:
1. `wxCAPTION` is not used by AUI, is it? So how could the check for this symbol fix an AUI problem?
2. It's not up to me or anyone else, it's up to whether it works better with that check or not. In conclusion, that code should be tested with a `wxFrame`, with and without caption, and confirm that mouse events get the correct position, as opposed to the code not containing that condition (and if it's still the case, maybe explain in the process, why such a condition is still necessary).

Reply:
Apologies as my colleague added the `wxCAPTION` section so I'm not 100% sure, but I assumed that AUI did make use of a caption for its title bar, which was not being taken into account and was the reason the AUI Panes could not be dragged by their title bar to dock and undock them.

I'm afraid I don't have the time to run additional tests on frames with and without captions, I can only say that we haven't seen any problems with frames that don't have captions elsewhere. That's why I would suggest potentially removing the wxCAPTION section if we can't confirm all scenarios have the correct position right now.